upgrade_testing/data/upgrade

```diff
@@ -310,7 +310,7 @@ function do_normal_upgrade() {
     # Allow upgrade from lts to non-lts if there's not lts to upgrade to
     local version
     if grep '^Prompt=lts' /etc/update-manager/release-upgrades; then
-        version=$(do-release-upgrade -d -p -c | awk '/New release/ {print $3}' | tr -d \')
+        version=$(do-release-upgrade -d -c | awk '/New release/ {print $3}' | tr -d \')
```

Comment:
Should we keep the `-p` instead? If that code executes then the system is an LTS (because it has `Prompt=lts`). The code checks if there's another LTS to upgrade to in a LTS1-to-LTS2 path. That upgrade will never happen with `-d`, because when an LTS2 is the devel release there's always a supported interim release between LTS1 and LTS2.

The `-p` is useful when LTS2 is out but LTS1-to-LTS2 upgrades are not enabled yet.

Reply:
I was wrong here: `do-release-upgrade -d` *does* upgrade LTS1-to-LTS2 when LTS2 is a devel release. That's because it will be listed on https://changelogs.ubuntu.com/meta-release-lts-development. Non-LTS devel releases will *not* be listed there, so `-d` will work fine even when there's a devel release newer than LTS2. Thanks @murraybd for clarifying.

Checking both `-d` and `-p` (separately) would still make sense should we want to test the LTS1-to-LTS2-to-LTS3 upgrade path *in a single upgrade profile* (e.g. bionic-to-jammy), which is something the tool supports (doing multiple upgrades in a row). Having only `-d` won't detect the `LTS1-to-LTS2` path.